pkg/pool-weighted/contracts/smart/WeightCompression.sol

```diff
@@ -23,6 +23,7 @@ import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
  */
 library WeightCompression {
     uint256 private constant _UINT31_MAX = 2**(31) - 1;
+    uint256 private constant _UINT32_MAX = 2**(32) - 1;
```

Comment:
Use `type(uint32).max` instead?

Reply:
We seem to be going back and forth on this. That's what it was, and then somebody on SC wanted it this way. I guess for consistency with the rest of the codebase - mainly WordCodec - that does it this way

Reply:
But if that's the case then why use type(uint64).max below?

Reply:
Oversight (fixed in a later branch) - will backport here to resolve confusion

Reply:
`type(uint32).max` is the better choice - but `uintX` is only defined for multiples of 8, so `uint31` does not exist unfortunately.

Reply:
Ok... so put it back to using max, except for 31